### ISSUE TITLE
[Assistive tech and device support] Alert and status messages aren't announced without receiving focus. (11.22.2)

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#34869]
#### VFS team name
Claims & Appeals

#### VFS product name
HLR V2

#### Point of Contact/Reviewers
Brian DeConinck (@briandeconinck) - Accessibility

---

### Platform Issue
Alert and status messages aren't announced without receiving focus.

### Issue Details
Auto-saving messages are not announced to screen reader users. Since the instructions indicate that auto-saving will be happening this isn't critical, but it's still information that should be made available to users as it happens.

### VA.gov Experience Standard
[Category Number 11, Issue Number 22](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Other References
WCAG SC 4.1.3_AA

---

### Platform Recommendation
Use aria-live (set to "polite") to announce when changes have been auto-saved. 

### VFS Team Tasks to Complete

- [ ] Comment on the ticket if there are questions or concerns
- [ ] Close the ticket when the issue has been resolved or validated by your Product Owner. If a team has additional questions or needs Platform help validating the issue, please comment in the ticket.

**Comment by Mottie** [comment]
I reported this problem to the design system team. It may not be ideal to announce the save since it is triggered every time the Veteran types a letter or clicks on a form element that changes the data. I recall Trevor and/or @joshkimux speaking about this and we opted to not announce this message. Instead the paragraph below the "Step X of X" includes a message "Your application will be saved on every change...". I would like to be included in any discussions on this topic.

**Comment by briandeconinck** [comment]
Thanks for the background. I believe @joshkimux is still out of the office but once he's back we can discuss and decide if it's still worth recommending any changes. For things like this it's always a tricky balance between [provide enough information to the user] and [don't bother the user with things they don't need to hear].

**Comment by joshkimux** [comment]
@briandeconinck yes, this was a tricky one and I think this is a good opportunity to revisit how auto saving messages are handled (from my goldfish memory, I think the last time we worked through this we talked about ways to address this more on the design front and less on the engineering end of things). 

I'll schedule some time for us to sync next week when I'm back from PTO @briandeconinck and @Mottie 🙇

**Comment by joshkimux** [comment]
After discussing with @Mottie @briandeconinck @AngelaFowler82 , we've decided to recommend using `aria-describedby` to link a portion of the auto save message to the continue button which:

- Better ensures screen reader users will _not_ miss the auto save message (which is located below the continue button)
- Does not change the visual design of the page
- Is flexible to the level of screen reader use (beginners are more likely to hear the auto-save message whereas advanced users won't be bothered by it)

<img width="1615" alt="Screen Shot 2022-01-19 at 9 27 29 AM" src="https://user-images.githubusercontent.com/14154792/150150705-b1e85332-0d9e-4f42-854b-54ed8cf185ff.png">

Would love help figuring out if this is something the design system team can help out with @k80bowman 🙏

**Comment by k80bowman** [comment]
@joshkimux it looks like our [progressButton](https://design.va.gov/storybook/?path=/docs/components-buttons-progressbutton--default) component has the capability of accepting an aria-describedby value. Does that meet your needs?

**Comment by joshkimux** [comment]
@k80bowman yes! That's just what we need 😄

**Comment by humancompanion-usds** [comment]
I will document this when we add Auto Save as a component which is tracked here:
https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1466
